h/models/group.py

```diff
@@ -214,13 +207,24 @@ def __acl__(self):
         if self.creator:
             # The creator of the group should be able to update it
             terms.append((security.Allow, self.creator.userid, 'admin'))
-            # auth_clients that have the same authority as this group
-            # should be allowed to update it
-            terms.append((security.Allow, authority_principal, 'admin'))
             terms.append((security.Allow, self.creator.userid, 'moderate'))
             # The creator may update this group in an upsert context
             terms.append((security.Allow, self.creator.userid, 'upsert'))
 
+        # This authority principal may be used to grant auth clients
```

Comment:
I consolidated all of the AuthClient-related principals in one spot for easier maintenance.